addons/website/static/src/js/content/website_root.js

```diff
@@ -82,6 +94,34 @@ var WebsiteRoot = publicRootData.PublicRoot.extend({
         }
         return registry;
     },
+    /**
+     * Toggles the fullscreen mode.
+     *
+     * @private
+     * @param {Boolean} state toggle fullscreen on/off (true/false)
+     */
+    async _toggleFullscreen(state) {
+        this.isFullscreen = state;
+        document.body.classList.toggle('o_fullscreen', this.isFullscreen);
+        document.body.style.overflowX = 'hidden';
+        let resizing = true;
+        window.requestAnimationFrame(function resizeFunction() {
+            window.dispatchEvent(new Event('resize'));
+            if (resizing) {
+                window.requestAnimationFrame(resizeFunction);
+            }
+        });
+        function onTransitionEnd(ev) {
+            if (ev.target === document.body && ev.propertyName === 'padding-top') {
+                resizing = false;
+                document.body.style.overflowX = '';
+                document.body.removeEventListener('transitionend', onTransitionEnd);
+            }
+        }
+        document.body.addEventListener('transitionend', onTransitionEnd);
+        // Safeguard in case the transitionend event doesn't trigger for whatever reason.
+        window.setTimeout(() => onTransitionEnd({target: document.body, propertyName: 'padding-top'}), 500);
```

Comment:
Can't you split the onTransitionEnd method so that here you don't have to add target and propertyName by hand? So that:
```suggestion
        window.setTimeout(() => stopResizing(), 500);
```